shotgun_api3/lib/mockgun/mockgun.py

```diff
@@ -625,31 +641,29 @@ def _get_field_from_row(self, entity_type, row, field):
                             raise ShotgunError("Invalid deep query field %s.%s" % (entity_type, field))
 
                         # Skips entities that are not of the requested type.
-                        if linked_row["type"] != entity_type2:
+                        if linked_row["type"] != entity_type:
                             continue
 
                         entity = self._db[linked_row["type"]][linked_row["id"]]
 
-                        sub_field_value = self._get_field_from_row(entity_type2, entity, field3)
+                        sub_field_value = self._get_field_from_row(entity_type, entity, rfield)
                         values.append(sub_field_value)
                     return values
                 # not multi entity, must be entity.
+                elif field_value is None:
```

Comment:
Did you undo the code in `find` as well? I meant that you should undo the changes from `_get_field_from_row` method. I've pushed a branch that restores the original code from `_get_field_from_row` and injects the two lines. You can see the diff with `master` [here](https://github.com/shotgunsoftware/python-api/compare/squeeze_fixes_update?expand=1).

When running the tests on that branch, everything still passes.